ui/src/app/job-details/failures/failures-tab/failures-tab.component.html

```diff
@@ -0,0 +1,49 @@
+<mat-table #table [dataSource]="dataSource">
+  <!-- Task name column -->
+  <ng-container matColumnDef="name">
+    <mat-header-cell *matHeaderCellDef> Task Name </mat-header-cell>
+    <mat-cell *matCellDef="let f" class="failure-task-name">{{ f.taskName }}</mat-cell>
+  </ng-container>
+
+  <!-- Failure message column -->
+  <ng-container matColumnDef="message">
+    <mat-header-cell *matHeaderCellDef> Failure Message </mat-header-cell>
+    <mat-cell *matCellDef="let f" class="failure-message">{{ f.failure }}</mat-cell>
+  </ng-container>
+
+  <!-- Outside links column -->
+  <ng-container matColumnDef="links">
+    <mat-header-cell *matHeaderCellDef> Log Files </mat-header-cell>
+    <mat-cell *matCellDef="let f" class="failure-links">
+      <clr-tooltip *ngIf="getResourceUrl(f.stdout)">
+        <a class="log-item" href="{{ getResourceUrl(f.stdout) }}">
+          <clr-icon clrTooltipTrigger shape="file" size="24"></clr-icon>
+          <clr-tooltip-content clrPosition="top-left" clrSize="xs" *clrIfOpen>
+            <span>stdout log</span>
+          </clr-tooltip-content>
+        </a>
+      </clr-tooltip>
+      <clr-icon *ngIf="!getResourceUrl(f.stdout)" shape="file" size="24" style="color: #ccc;"></clr-icon>
```

Comment:
Per my other comment about logs on `FailureMessage` this should extract the correct log file out of the `job` or `task` (if a taskName is present).